src/util/key.rs

```diff
@@ -35,7 +35,7 @@ pub struct PublicKey {
 
 impl PublicKey {
     /// Write the public key into a writer
-    pub fn write_into<W: io::Write>(&self, writer: &mut W) {
+    pub fn write_into<W: io::Write>(&self, mut writer: W) {
```

Comment:
What does this change exactly? I never saw this to be honest..

Reply:
It lets you pass a `&mut [u8]` to the function.

Reply:
See also https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#generic-readerwriter-functions-take-r-read-and-w-write-by-value-c-rw-value